include/libdnf/rpm/repo.hpp

```diff
@@ -118,262 +109,349 @@ struct Repo {
+    /// Will throw exception if Repo has no mirror or baseurl set or if Repo type is unsupported.
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.verify()
+    void verify() const;
 
-    Repo & operator =(Repo && repo) = delete;
+    /// Returns pointer to the repository configuration
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getConfig()
+    ConfigRepo * get_config() noexcept;
 
-    void setCallbacks(std::unique_ptr<RepoCB> && callbacks);
+    /// Returns repository id
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getId()
+    const std::string & get_id() const noexcept;
 
-    /**
-    * @brief Verify repo object configuration
-    *
-    * Will throw exception if Repo has no mirror or baseurl set or if Repo type is unsupported.
-    */
-    void verify() const;
-    ConfigRepo * getConfig() noexcept;
-    const std::string & getId() const noexcept;
+    /// Enable the repository
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.enable()
     void enable();
+
+    /// Disable the repository
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.disable()
     void disable();
-    bool isEnabled() const;
-    bool isLocal() const;
-    /**
-    * @brief Initialize the repo with metadata
-    *
-    * Fetches new metadata from the origin or just reuses local cache if still valid.
-    *
-    * @return true if fresh metadata were downloaded, false otherwise.
-    */
+
+    /// Return whether the repository is enabled.
+    /// @return true if enabled
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.isEnabled()
+    bool is_enabled() const;
+
+    /// Return whether the repository is local.
+    /// @return true if local
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.isLocal()
+    bool is_local() const;
+
+    /// Initialize the repo with metadata
+    /// Fetches new metadata from the origin or just reuses local cache if still valid.
+    /// @return true if fresh metadata were downloaded, false otherwise.
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.load()
     bool load();
-    bool loadCache(bool throwExcept);
-    void downloadMetadata(const std::string & destdir);
-    bool getUseIncludes() const;
-    void setUseIncludes(bool enabled);
-    bool getLoadMetadataOther() const;
-    void setLoadMetadataOther(bool value);
-    int getCost() const;
-    int getPriority() const;
-    std::string getCompsFn();  // this is temporarily made public for DNF compatibility
+
+    /// Loads metadata from local cache
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.loadCache(bool throwExcept)
+    bool load_cache(bool throw_except);
+
+    /// Downloads all metadata
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.downloadMetadata(const std::string & destdir)
+    void download_metadata(const std::string & destdir);
+
+    /// Returns whether the using of "includes" is enabled
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getUseIncludes()
+    bool get_use_includes() const;
```

Comment:
What kind of includes do you mean? rpm includes or config includes?

Reply:
If enabled, only packages listed in the "includepkgs" will be used from the repository.
I will improve docs.